tensorboard/webapp/hparams/_redux/hparams_selectors.ts

```diff
@@ -178,3 +184,51 @@ export const getExperimentsHparamsAndMetricsSpecs = createSelector(
     );
   }
 );
+
+export const getDashboardHparamsAndMetricsSpecs = createSelector(
+  getHparamsState,
+  (state: HparamsState) => {
+    return state.dashboardSpecs;
+  }
+);
+
+export const getDashboardRunsToHparamsAndMetrics = createSelector(
+  getHparamsState,
+  (state): RunToHparamsAndMetrics => {
+    const runToHparamsAndMetrics: RunToHparamsAndMetrics = {};
+
+    for (const sessionGroup of state.dashboardSessionGroups) {
+      const hparams: HparamValue[] = Object.entries(sessionGroup.hparams).map(
+        (keyValue) => {
+          const [hparam, value] = keyValue;
+          return {name: hparam, value};
+        }
+      );
+
+      for (const session of sessionGroup.sessions) {
+        runToHparamsAndMetrics[session.name] = {
+          metrics: [],
+          hparams,
+        };
+
+        for (const metricValue of session.metricValues) {
+          const runId = metricValue.name.group
```

Comment:
On the other hand, without metric values from the /session_groups requests, you wouldn't be able to calculate runIds in this way - you'd have to find a different approach. Perhaps get a list of runs for each experiment (I believe you get this elsewhere, too) and then determine which session groups they belong to?

Reply:
From an offline discussion:
 * We will calculate this mapping using existing data on the front end (probably from the /runs response) paired with the Session.name value.